teps/0089-spire-support.md

```diff
@@ -0,0 +1,215 @@
+  - [Non-Goals](#non-goals)
+- [SPIRE Concepts](#spire-concepts)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Requesting an SVID/Signature from the Controller Image](#requesting-an-svidsignature-from-the-controller-image)
+  - [Mounting SPIRE into the Tekton Controller](#mounting-spire-into-the-tekton-controller)
+- [Test Plan](#test-plan)
+- [Implementation Plan](#implementation-plan)
+- [Alternatives](#alternatives)
+  - [Kubernertes Service Account Token Volume Projection](#kubernertes-service-account-token-volume-projection)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP covers integrating Tekton and Tekton Chains with [SPIFFE/SPIRE](https://spiffe.io/), which would provide a more secure supply chain for Tekton users. It would also guarantee [non-falsifiable provenance](https://slsa.dev/requirements#non-falsifiable), which is a requirement for [SLSA Level 3](https://slsa.dev/levels).
+With this integration, Tekton will be one step closer to SLSA Level 3 compliance.
+
+Currently, Tekton Chains observes Tekton and waits for TaskRuns to complete.
+Once it sees that a TaskRun has completed it tries to sign any artifacts that were built and also generates provenance.
+
+There are a couple issues with this:
+1. Tekton Chains has no way to verify that a TaskRun in the cluster was created by Tekton
+2. Tekton Chains has no way to verify that the TaskRun it received wasn't modified by anybody other than Tekton, during or after execution
+3. Tekton Pipelines can't verify that the results it reads weren't modified
+
+This is where SPIRE comes in!
+SPIRE can be used to request signatures and certificates (SVIDs) for a given workload (this workload could be a result, or in our case an entire TaskRun yaml).
+SPIRE can be configured to only issue SVIDs to certain k8s workloads with a SPIRE [k8s Workload Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md).
+In thise case, we'll set up SPIRE to only issue SVIDs to:
+* The Tekton controller
+* Pods created by the Tekton controller (e.g. pods running the entrypointer image)
+
+This configuration will let us ensure that all Tasks were created by Tekton and that Task results weren't modified.
+
+The basic flow would look like this:
+1. Tekton Pipelines receives a TaskRun config, and generates the Pod for with SPIRE mounted in to the Pod
+1. Tekton Pipelines stores the state of the TaskRun in memory, and every time the status updates it follows this loop:
+* Compare the current TaskRun to the state stored in memory
+* If the TaskRun has been changed, don't request an SVID from SPIRE
+* If the TaskRun has not been changed, then update the TaskRun status and store the new state in memory
+1. The entrypointer image requests an SVID and signature for all results in the Pod
```

Comment:
i think it could be worth being specific about when this would happen - results will be emitted as containers execute, so I'm guessing the entrypointer would be signing each result before writing it to the termination message (or wherever the result ends up)